File: app/src/main/java/com/davidbronn/fragmentrevision/activity_to_fragment/MainActivity.java
package com.davidbronn.fragmentrevision.activity_to_fragment;

import android.databinding.DataBindingUtil;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.TextView;

import com.davidbronn.fragmentrevision.R;
import com.davidbronn.fragmentrevision.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity
{
    private ActivityMainBinding binding;
    private TextChangedEvent changedEvent;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        String TAG = "SAMPLE";
        FirstFragment firstFragment = new FirstFragment();
        getSupportFragmentManager().beginTransaction().replace(R.id.frameLayout, firstFragment, TAG).commit();
        changedEvent = firstFragment;

        binding.editText.setOnEditorActionListener((v, actionId, event) ->
        {
            if(actionId == EditorInfo.IME_ACTION_DONE)
            {
                changedEvent.onTextChanged(binding.editText.getText().toString());
                return true;
            }
            return false;
        });
    }
}

File: app/src/main/java/com/davidbronn/fragmentrevision/activityToFragment/MainActivity.kt
package com.davidbronn.fragmentrevision.activityToFragment

import android.os.Bundle
import androidx.core.widget.addTextChangedListener
import androidx.databinding.DataBindingUtil
import com.davidbronn.fragmentrevision.R
import com.davidbronn.fragmentrevision.databinding.ActivityMainBinding
import com.davidbronn.fragmentrevision.utils.BaseActivity
import com.davidbronn.fragmentrevision.utils.TextChangedEvent

class MainActivity : BaseActivity() {

    private var binding: ActivityMainBinding? = null
    private var changedEvent: TextChangedEvent? = null

    private val fragment by lazy { FirstFragment() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        addFragment(R.id.frameLayout, fragment)
        changedEvent = fragment

        binding?.editText?.addTextChangedListener {
            changedEvent?.onTextChanged(it.toString())
        }
    }
}
